src/us/kbase/auth2/lib/Password.java

```diff
@@ -42,4 +61,37 @@ public void clear() {
 			password[i] = '0';
 		}
 	}
+	
+	/** Check for password validity (length and strength requirements)
+	 * @throws IllegalPasswordException if the password is not valid
+	 */
+	public void checkValidity() throws IllegalPasswordException {
+		// check length requirements
+		if(password.length > MAX_PASSWORD_LENGTH) {
+			throw new IllegalPasswordException("Password exceeds max length ("+
+												MAX_PASSWORD_LENGTH+")");
+		}
+		
+		// check strength requirement
+		Strength strength = new Zxcvbn().measure(new String(password));
```

Comment:
any way you know of to mitigate that risk?  We could explicitly create and then clear the string, but not sure if that actually clears out the memory.